src/parser/token.ml

```diff
@@ -136,7 +137,9 @@ type t =
   | T_EMPTY_TYPE
   | T_BOOLEAN_TYPE of bool_or_boolean
   | T_NUMBER_TYPE
+  | T_BIGINT_TYPE
   | T_NUMBER_SINGLETON_TYPE of { kind: number_type; value: float; raw: string }
+  | T_BIGINT_SINGLETON_TYPE of { kind: number_type; value: float; raw: string }
```

Comment:
Maybe this should also `s/value/approx_value/`? I feel less strongly here than in the ast, since the lexer data structures are far less commonly used.